plugins/woocommerce/includes/wc-order-functions.php

```diff
@@ -727,7 +727,22 @@ function wc_restock_refunded_items( $order, $refunded_line_items ) {
 		}
 
 		/* translators: 1: product ID 2: old stock level 3: new stock level */
-		$order->add_order_note( sprintf( __( 'Item #%1$s stock increased from %2$s to %3$s.', 'woocommerce' ), $product->get_id(), $old_stock, $new_stock ) );
+		$restock_note = sprintf( __( 'Item #%1$s stock increased from %2$s to %3$s.', 'woocommerce' ), $product->get_id(), $old_stock, $new_stock );
+		
+		/**
+		 * Allow the restock note to be modified.
+		 *
+		 * @since 6.2.0
```

Comment:
Looks good but since we're now past code freeze for 6.2 this will need to go into 6.3 instead. So please apply the following change and I'll approve:

```suggestion
		 * @since 6.3.0
```